src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -774,28 +775,46 @@ internal async Task SearchPackagesAndRefreshUpdateCountAsync(string searchText,
                 }
                 else
                 {
-                    RefreshAvailableUpdatesCount();
+                    RefreshInstalledAndUpdatesTabs();
                 }
             }
         }
 
-        private void RefreshAvailableUpdatesCount()
+        private void RefreshInstalledAndUpdatesTabs()
         {
             NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
+                _topPanel._labelInstalled.ShowWarning = false;
                 _topPanel._labelUpgradeAvailable.Count = 0;
                 var loadContext = new PackageLoadContext(ActiveSources, Model.IsSolution, Model.Context);
                 var packageFeed = await CreatePackageFeedAsync(loadContext, ItemFilter.UpdatesAvailable, _uiLogger);
                 var loader = new PackageItemLoader(
                     loadContext, packageFeed, includePrerelease: IncludePrerelease);
+                var metadataProvider = CreatePackageMetadataProvider(loadContext);
 
-                // cancel previous refresh update count task, if any
+                // cancel previous refresh tabs task, if any
                 // and start a new one.
                 var refreshCts = new CancellationTokenSource();
                 Interlocked.Exchange(ref _refreshCts, refreshCts)?.Cancel();
 
+                // Update installed tab warning icon
+                var installedPackages = await loadContext.GetInstalledPackagesAsync();
+                var installedPackageDeprecationMetadata = await Task.WhenAll(
+                    installedPackages.Select(p => GetPackageDeprecationMetadataAsync(p, metadataProvider, refreshCts.Token)));
+
+                var installedDeprecatedPackagesCount = installedPackageDeprecationMetadata.Count(d => d != null);
+                var hasInstalledDeprecatedPackages = installedDeprecatedPackagesCount > 0;
+                _topPanel._labelInstalled.ShowWarning = hasInstalledDeprecatedPackages;
+                _topPanel._labelInstalled.WarningToolTip = hasInstalledDeprecatedPackages
+                    ? string.Format(
+                        CultureInfo.CurrentCulture,
+                        Resx.Resources.Label_Installed_DeprecatedWarning,
+                        installedDeprecatedPackagesCount)
+                    : null;
```

Comment:
Is any work now being performed on the UI thread which was not performed on the UI thread prior to your change?

Reply:
Looks like you were correct that the registration information was being fetched on the UI thread here. I have updated it so that it switches off the thread to fetch the information. The updates tab does a similar thing, but I missed it because it switches off the thread inside the `loader.GetAllPackagesAsync` call.